monitor/verifier.go

```diff
@@ -59,6 +69,92 @@ func (m *EthMonitor) verifySubscribe(change sync.Change) bool {
 	return subscription.Deposit.BigInt().Cmp(deposit) == 0
 }
 
+func (m *EthMonitor) verifyRequest(change sync.Change) bool {
+	var request subscribe.Request
+	m.operator.CliCtx.Codec.MustUnmarshalBinaryBare(change.Data, &request)
+	log.Infoln("Verify request", request)
+
+	var signedSimplexState chain.SignedSimplexState
+	err := proto.Unmarshal(request.SignedSimplexStateBytes, &signedSimplexState)
+	if err != nil {
+		log.Errorln("Failed to unmarshal signedSimplexStateBytes:", err)
+		return false
+	}
+
+	r, err := subscribe.GetRequest(m.operator.CliCtx, m.ethClient, &signedSimplexState)
+	if err != nil {
+		log.Errorln("Failed to get request through SignedSimplexStateBytes:", err)
+		return false
+	}
+
+	err = subscribe.VerifySignedSimplexStateSigs(request, signedSimplexState)
+	if err != nil {
+		log.Infoln("Failed to verify sigs:", err)
+		return false
+	}
+
+	return request.SeqNum > r.SeqNum && request.PeerFromIndex == r.PeerFromIndex &&
+		reflect.DeepEqual(request.ChannelId, r.ChannelId) && reflect.DeepEqual(request.PeerAddresses, r.PeerAddresses)
```

Comment:
use `bytes.Compare()`. Though reflect also works, it's better to use the more common and less advanced lib if it can do the job.

Reply:
request.PeerAddresses is []string, so I will only use `bytes.Compare() ` for `request.ChannelId `